### PR TITLE
feat(suite): mark potentially phishy zero-value transactions

### LINKED ISSUE #7278: Address poisoning attack: Label zero value transactions in TX detail
## Background

One type of scam is that anyone can send a transaction from any address with 0 amount without the private keys. Attackers will send the 0 amount to the address, which is almost the same as the address of e.g. Binance. The user won't realize it as the beginning and the end of these addresses look the same. User will then just copy the address from TX history and send funds to that (wrong) address.

## Proposed changes

Label zero value transactions from the TX list and detail as scammy.

## Concepts

image (4)

Tooltip and red banner to contain link to [KB article](https://trezor.io/support/a/address-poisoning-attacks)

Screenshot 2023-01-13 at 16 51 12

image (5)(1)

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* mark potentially phishy zero-value transactions

## Related Issue

Resolve [#7278](https://github.com/trezor/trezor-suite/issues/7278)

## Screenshots:

image
image

### CODE REVIEW

File: suite-common/suite-utils/src/__tests__/antiFraud.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import type { TokenTransfer } from '@trezor/blockchain-link';
+import type { WalletAccountTransaction } from '@suite-common/wallet-types';
+
+import { getIsZeroValuePhishing } from '../antiFraud';
+
+describe('antifraud utils', () => {
+    test('detects potential zero-value hoshing transactions', () => {
```
Comment (komret) [inline suite-common/suite-utils/src/__tests__/antiFraud.test.ts]: hoshing?

File: packages/suite/src/components/suite/modals/TransactionDetail/index.tsx [hunk 2]
```diff
@@ -22,9 +24,19 @@ const StyledModal = styled(Modal)`
...
-const Wrapper = styled.div`
-    width: 100%;
-    flex-direction: column;
+const PhishingBanner = styled.div`
+    margin-bottom: 6px;
+    padding: 6px 0;
+    border-radius: 8px;
+    background: ${({ theme }) => theme.BG_RED};
+    color: ${({ theme }) => theme.TYPE_WHITE};
+    font-size: 14px;
```
Comment (komret) [inline packages/suite/src/components/suite/modals/TransactionDetail/index.tsx]: `font-size` is repeated below.

File: packages/suite/src/components/wallet/TransactionItem/components/TargetAddressLabel.tsx [hunk 3]
```diff
@@ -28,7 +35,7 @@ export const TargetAddressLabel = ({ target, type, accountMetadata }: TargetAddr
     }
 
     return (
-        <TruncatedSpan>
+        <TruncatedSpan isBlurred={isZeroValuePhishing}>
```
Comment (komret) [inline packages/suite/src/components/wallet/TransactionItem/components/TargetAddressLabel.tsx]: This does not apply, the relevant transaction items use this component instead: `TokenTransferAddressLabel`.

**Review by komret — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/wallet/TransactionItem/components/TargetAddressLabel.tsx (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/components/wallet/TransactionItem/components/TargetAddressLabel.tsx]: [743062e](https://github.com/trezor/trezor-suite/pull/7391/commits/743062eb2a78766ab992206c2d807d843b570601)

File: packages/suite/src/components/suite/modals/TransactionDetail/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/TransactionDetail/index.tsx]: [743062e](https://github.com/trezor/trezor-suite/pull/7391/commits/743062eb2a78766ab992206c2d807d843b570601)

File: suite-common/suite-utils/src/__tests__/antiFraud.test.ts (hunk 1 above)
Comment (dahaca) [reply suite-common/suite-utils/src/__tests__/antiFraud.test.ts]: [743062e](https://github.com/trezor/trezor-suite/pull/7391/commits/743062eb2a78766ab992206c2d807d843b570601)

**Review by komret — APPROVED** [review]
Code is fine, but I have some doubts regarding the logic for marking transactions as discussed in #7278, @Hannsek.

**Review by Horlabrainmoore — COMMENTED** [review]
akto-source-code-analyserContainers Images

aktosecurity/akto-puppeteer-replay:doom_latest

hotavneesh/eclipse-jdtls:latest

aktosecurity/source-code-analyser:a-1703-merge

Displaying only the first 5 entries

https://payload-marketing.moonpay.com/api/media/file/st8e72d7nvj-5rFBKt3kQYYBlfmSfPSJeJ```[https://links2.cointracker.io/u/click?_t=52e482ca0b874b77945a2bb7da40c25d&_m=f55f4270b3d1431e949f501a121fc854&_e=bZxStQVB2iwB-tRYhVYGIq4bHtr1RDEkCQvRwwMvdr_H7KSUESg3ldbPHMGzK5jgwRnUnoDeyohUcinEigUA-udd3gjVj5mPAb6fiJSNwnrmi6YL5rKEAw-2CnXxhXwsOcD0jzCAXjQZ6l2cBXGdMtWCKzAZJrhwm6ki2IpeHnOsWYdOPdEClzP7r3gscUW7M_c0sJMh60VahT1urX4WozeF_8CvPYDRcNl0rN4T6Vqv7IjWF4jucxfxAf_Eut86uABMpxtx4OCBp3K0SKPFGV4FJFssiat46CRYz8M5-Vq-Bciz34ZrD_rtX_-mRfot```{"Name":"Evan Goldner","Credit_Card_Number":"4632655004713664","Issuer":"visa","Expiry_Date":"01/29","cvv":"461"},{"Name":"Hector Nicolas","Credit_Card_Number":"4424720220995867","Issuer":"visa","Expiry_Date":"03/28","cvv":"665"},{"Name":"Ollie Franecki","Credit_Card_Number":"4845043884072993","Issuer":"visa","Expiry_Date":"10/25","cvv":"828"},{"Name":"Roderick Huel","Credit_Card_Number":"4888484722780959","Issuer":"visa","Expiry_Date":"08/28","cvv":"118"},{"Name":"Marcus Senger","Credit_Card_Number":"4685824515948716","Issuer":"visa","Expiry_Date":"04/28","cvv":"619"},{"Name":"Mark Witting","Credit_Card_Number":"4182014616022235","Issuer":"visa","Expiry_Date":"03/25","cvv":"665"},{"Name":"Robin Pacocha","Credit_Card_Number":"4488163020354717","Issuer":"visa","Expiry_Date":"05/27","cvv":"220"},{"Name":"Amber Pfannerstill","Credit_Card_Number":"4089217241674080","Issuer":"visa","Expiry_Date":"04/28","cvv":"855"},{"Name":"Dr. Paul Hammes","Credit_Card_Number":"4889334523597012","Issuer":"visa","Expiry_Date":"11/28","cvv":"125"},{"Name":"Emanuel Schaefer","Credit_Card_Number":"4352839403090043","Issuer":"visa","Expiry_Date":"05/29","cvv":"814"}] @horlabrainmoorebtc04t-1735516333301.atlassian.Net

**Pull request merged**